plugins/tangle/persisters.go

```diff
@@ -75,54 +75,47 @@ func addressPersisterSubmit(address trinary.Hash, transactionHash trinary.Hash)
 	addressPersisterWorkerPool.Submit(&tangle.TxHashForAddress{Address: address, TxHash: transactionHash})
 }
 
-// Unconfirmed Tx persister
-func configureUnconfirmedTransactionPersister() {
+// FirstSeen Tx persister
+func configureFirstSeenTransactionPersister() {
 
-	unconfirmedTxWorkerPool = batchworkerpool.New(func(tasks []batchworkerpool.Task) {
+	firstSeenTxWorkerPool = batchworkerpool.New(func(tasks []batchworkerpool.Task) {
 
-		var operations []*tangle.UnconfirmedTxHashOperation
+		var operations []*tangle.FirstSeenTxHashOperation
 		for _, task := range tasks {
-			operations = append(operations, task.Param(0).(*tangle.UnconfirmedTxHashOperation))
+			operations = append(operations, task.Param(0).(*tangle.FirstSeenTxHashOperation))
 		}
 
-		err := tangle.StoreUnconfirmedTxHashOperations(operations)
+		err := tangle.StoreFirstSeenTxHashOperations(operations)
 		if err != nil {
 			panic(err)
 		}
 
 		for _, task := range tasks {
 			task.Return(nil)
 		}
-	}, batchworkerpool.BatchCollectionTimeout(unconfirmedTxBatchCollectionTimeout), batchworkerpool.BatchSize(unconfirmedTxBatchSize), batchworkerpool.WorkerCount(unconfirmedTxWorkerCount), batchworkerpool.QueueSize(unconfirmedTxQueueSize), batchworkerpool.FlushTasksAtShutdown(true))
+	}, batchworkerpool.BatchCollectionTimeout(firstSeenTxBatchCollectionTimeout), batchworkerpool.BatchSize(firstSeenTxBatchSize), batchworkerpool.WorkerCount(firstSeenTxWorkerCount), batchworkerpool.QueueSize(firstSeenTxQueueSize), batchworkerpool.FlushTasksAtShutdown(true))
 }
 
-func runUnconfirmedTransactionPersister() {
+func runFirstSeenTransactionPersister() {
 
 	notifyNewTx := events.NewClosure(func(transaction *hornet.Transaction, firstSeenLatestMilestoneIndex milestone_index.MilestoneIndex, latestSolidMilestoneIndex milestone_index.MilestoneIndex) {
-		unconfirmedTxWorkerPool.Submit(&tangle.UnconfirmedTxHashOperation{
-			TxHash:                        transaction.GetHash(),
-			FirstSeenLatestMilestoneIndex: firstSeenLatestMilestoneIndex,
-		})
-	})
-
-	notifyConfirmedTx := events.NewClosure(func(transaction *hornet.Transaction, msIndex milestone_index.MilestoneIndex, confTime int64) {
-		unconfirmedTxWorkerPool.Submit(&tangle.UnconfirmedTxHashOperation{
-			TxHash:    transaction.GetHash(),
-			Confirmed: true,
-		})
+		if !transaction.IsRequested() {
```

Comment:
Make a comment above this line explaining why requested transactions do not have have to be persisted in the firstSeenMilestone database. (obviously they should get confirmed, that's the reason, and therefore auto. be approvees of a milestone when pruning)